pkg/middleware/csrf.go

```diff
@@ -4,10 +4,11 @@ import (
 	"errors"
 	"net/http"
 	"net/url"
-	"strings"
+
+	"github.com/grafana/grafana/pkg/util"
 )
 
-func CSRF(loginCookieName string) func(http.Handler) http.Handler {
+func CSRF(loginCookieName, defaultPort string) func(http.Handler) http.Handler {
```

Comment:
Not sure the `defaultPort` needs to be passed since only used for setting the default and can be hardcoded inside the `CSRF` function maybe?

Reply:
I see for the check we don't use the Port property, so in that sense it makes sense (passing and do nothing could be misleading).

But I'm not clear on all the cases here, for example is sending `Host: "locatlhost:80", and `Origin: "http://localhost" valid if it is on port is 80 (other the other way with schema on origin still)?

If we do want to ignore the port entirely, then can add a comment and some tests that demonstrate a port mismatch is allowed.

Reply:
As I understood it we don't care about the port. See also @DanCech comment below.